src/tools/compiletest/src/runtest.rs

```diff
@@ -3238,6 +3241,244 @@ impl<'test> TestCx<'test> {
         fs::remove_dir(path)
     }
 
+    fn run_rmake_v2_test(&self) {
```

Comment:
This should be possible: instead of a full alternative directory, I could just make the distinction between `run-make` and `run-make-v2` tests the presence/absence of `Makefile` vs `__rmake.rs` and it would be a matter of just changing how tests are collected (I think).

Reply:
And migrating to use Rust makefiles should then just be replacing `Makefile` with `__rmake.rs`.

Reply:
I piggy-backed off of `run-make` and have `run-make` tests conditionally run in V2 instead of having an entire separate `run-make-v2` test mode and directory, based on the existence of `__rmake.rs` in the test's directory instead of `Makefile`. Migration from V1 to V2 should be as simple as replacing `Makefile` with `__rmake.rs` under `tests/run-make`. This should hopefully reduce the noise on git history with migrations.